v2/writer_test.go

```diff
@@ -131,3 +131,150 @@ func assertAddNodes(t *testing.T, adder format.NodeAdder, nds ...format.Node) {
 		assert.NoError(t, adder.Add(context.Background(), nd))
 	}
 }
+
+func TestReplaceRootsInFile(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		roots   []cid.Cid
+		wantErr bool
```

Comment:
nit: I'd make this a string and do a `strings.Contains`, to have a better idea of how each test case is expected to fail.

I know that comparing error strings is generally a bad thing, but if you're testing your own code, I think doing that is pretty reasonable - and not prone to breakages as long as you check for substrings rather than entire strings.